### PR TITLE
Add proxy support to vic-machine for image fetching

### LINKED ISSUE #2537: Add proxy support to the appliance
There is currently no persistent mechanism for configuring a proxy server. While /etc/sysconfig/proxy can be modified it will not persist across reboot.

I'm unsure how much code, if any, in the appliance is necessary to support this or whether the Go http stack handles this transparently.

If we have to provide different proxies for the different networks it will get involved, if only from a command line args perspective. The raw number of options to vic-machine is steadily bumping the importance of the manifest support.

This is at the request of a customer via @cormachogan

### PR DESCRIPTION
**Pull request by hickeng**
The fetcher component (imagec) requires support for a proxy in many
customer deployments. This adds the vic-machine glue for correctly
populating the environment variables of the personality so that fetcher
has them available.

This does NOT provide or configure proxy support for containers run by
this VCH.

Fixes #2537

### CODE REVIEW

File: vendor/github.com/urfave/cli/app.go [hunk 1]
```diff
@@ -467,6 +467,7 @@ func HandleAction(action interface{}, context *Context) (err error) {
 			switch r.(type) {
 			case error:
 				err = r.(error)
+				panic(err)
```
Comment (andrewtchin) [inline vendor/github.com/urfave/cli/app.go]: Did you mean to leave this?

File: cmd/vic-machine/create/create.go [hunk 2]
```diff
@@ -384,6 +387,23 @@ func (c *Create) Flags() []cli.Flag {
...
+			Value:       "",
+			Usage:       "An HTTPS proxy for use when fetching images, in the form https://fqdn_or_ip:port",
+			Destination: &c.httpsProxy,
+			Hidden:      true,
+		},
+
+		cli.StringFlag{
+			Name:        "http-proxy, hproxy",
+			Value:       "",
+			Usage:       "An HTTP proxy for use when fetching images, in the for http://fqdn_or_ip:port",
```
Comment (sgairo) [inline cmd/vic-machine/create/create.go]: "in the form"

File: vendor/github.com/urfave/cli/app.go (hunk 1 above)
Comment (hickeng) [inline vendor/github.com/urfave/cli/app.go]: no, no I did not. Wish there was a way to toggle this with --debug... it's a pain not having any idea where panic's come from.

**Comment by emlin** [comment]
lgtm 👍

File: cmd/vic-machine/create/create.go (hunk 2 above)
Comment (hickeng) [inline cmd/vic-machine/create/create.go]: done

**Comment by andrewtchin** [comment]
lgtm

**Comment by hickeng** [comment]
poke

**Pull request merged**